Kaoru setup gulp init state

KaoruDev opened the pull request:
### Objective:

Setup a take home repo to gain better insights on a candidate's technical skills set. This branch setups a `gulp` system that watches / compiles (sass / js ) / copys htmls / serves ( via browser-sync ). The system is a `submodule` pointing [here](https://github.com/kaorudev/jelly).

KaoruDev commented:
@noahgrant @zorahlea  would ❤️ some feedback please

noahgrant commented:
cool!  i'm going to upgrade to node 6 (with our build stuff) in the next day or two and then i'll be able to run this :)

noahgrant commented on README.md:
```diff
@@ -0,0 +1,75 @@
+     - Event Bindings / Delegations
+
+#### Restrictions:
+Please reframe from using additional libraries ( jQuery is already installed ).
```
(nit: `refrain`).  also extraneous spaces between parens?  also, if the below is true, why are also including jquery?

noahgrant commented on README.md:
```diff
@@ -0,0 +1,75 @@
+#### Problems to solve:
+- Finish the panel navigation so when a user clicks on a different product tab
+ the corresponding product panel appears.
+ - You're job is also to make this panel responsive. Can a user access the same
```
nit: `your`

noahgrant commented on README.md:
```diff
@@ -0,0 +1,75 @@
+
+#### Nice to haves:
+
+- Performant animations. I.e. leverage GPU vs CPU.
```
is this relevant?  where are the animations going to come in?  If each panel should slide in or fade in, maybe we should be explicit about that?  ie `Nice to haves: fade each panel in! (and make sure to do it performantly!)` I'd recommend seeing if they leverage the GPU without us needing to hint at it.

noahgrant commented on styles/pages/articles/_sections.scss:
```diff
@@ -0,0 +1,46 @@
+}
+
+.product-hero-container, .product-features-container {
+  width: 49%;
```
i think it's good practice to make anything that should be half space be an even `50%` when we can make it so (ie when we're not constrained by the something in the actual container).  same goes for making the `text-container` below `70%` instead of `69%`.

i think it reflects better on us, as well, when giving an assignment like this, because putting 69%, for example, kinda implies that we're hacking something together to make it work just right. elements that are inline block, when the markup is minified, should fit exactly side by side with no extra `inline-block` gap. so we can have a 50%/50% with no gap hacks, and can treat any inner spacing as padding, which is appropriate.

noahgrant commented:
without pulling this up locally, i think this seems like a really nice take-home exercise—a little javascript, but not too much, and a nice amount of CSS.  i'm wondering if we should also have them add their own markup somewhere so that we can see how semantic they are?  OR, we can make our markup riddled with errors or unsemantic-y-ness and ask them to fix it up?

another thing i think we should think about doing is installing modernizr and asking them to provide fallbacks to support any modern features they use, >= IE9.  what do you think?

KaoruDev commented:
@noahgrant address the things you pointed out. `Modernizer` sounds like a good idea! Adding another page to build might be too much. I for one would get stuck on how to `design` the page rather how I would write it. Although maybe we could take a screenshot of something on our home page and have them add it to the current page. Any ideas which section you'd like to see?

noahgrant commented:
that's a great idea ^^.  how about this, since we really should have them write some markup:

```
But one of the panels is empty!  The ____  panel is supposed to look like this:

<screenshot of the missing panel>

Please provide the relevant markup along with the styling to add back our missing panel!
```

zorahlea commented on README.md:
```diff
@@ -0,0 +1,79 @@
+
+
+#### Real Live Demo:
+[https://siftscience.com/chargebacks#features](https://siftscience.com/chargebacks#features)
```
This just redirects to https://siftscience.com/products/payment-fraud to be "backwards compatible." can we change this link to the current link?

zorahlea commented:
How long do you anticipate this taking? Without reading into this too much, this seems pretty overwhelming for a take home.

KaoruDev commented:
kk per our conversation during lunch I stripped the project down to the basics. I decided to _not_ include Modernizr because I'd like to leave it up to the candidate to choose the tools to how to solve the problems they encounter.

i.e. I added the requirement to support `IE10` how they decide to support it should be up to them. 

If you disagree we can chat about it more when Gary gets back.

noahgrant commented on README.md:
```diff
@@ -0,0 +1,57 @@
+  [marketing_assets/enterprise-grade-copy.txt](https://github.com/SiftScience/take-home-web-dev/blob/master/marketing-assets/enterprise-grade-copy.txt) and
+  designs in [images/enterprise-grade-design.png](https://github.com/SiftScience/take-home-web-dev/blob/master/images/enterprise-grade-design.png).
+  Please feel free to use placeholder images. Services like [placekitten](http://placekitten.com) or [lorempixel](http://lorempixel.com/) are great.
+  Font color is: `#416d81` and gray background is: `#f1f1f1`
```
is this necessary? they should be able to get this from the css, right?  or change it if they feel so inclined?

noahgrant commented on README.md:
```diff
@@ -0,0 +1,57 @@
+- Use of css specificity
+- Readable, maintable code. I.e. modularity, rememberable name choices
+(i.e. **not** `this-box`)
+- Support for IE10+
```
i really think we should bring back Modernizr and our old statement, but take out any reference to old browser support.  the reason is that we shouldn't expect them to have access to browserstack or old ie images—even though the images are free, they take awhile to setup and are a pain to test on.

additionally, i think that, with this statement, people will opt for old-school tactics (ie floats) that they know have wide browser support—but that's not entirely what we're interested in. by mentioning Modernizr and saying something more general like

```
'Feel free to use any modern CSS features you like, but make sure to provide fallbacks for browsers that might not support them—we have Modernizr included for your convenience.'
```
we can encourage them to use, for example, flexbox, that they'd gain bonus points for knowing, and even more bonus points for providing fallbacks for, and that they might have avoided using at all if they were just trying to play it safe.

noahgrant commented on README.md:
```diff
@@ -0,0 +1,57 @@
+- Event Bindings / Delegations
+
+#### Restrictions:
+Please do not from using css frameworks like Bootstrap or Foundation. You're welcome to use any other tools to aid you, though it is **not required**.
```
grammarz ^^.  Also, what other tools do you have in mind here?  If you're referring to pre-processors, let's be more explicit about that here, like saying:

`You're welcome to use any preprocessors you wish, like LESS or SASS, but you'll need to include your own build step.`

we can also link to where to install those binaries ^^, or have an addendum at the bottom of this README that we link to about how to add a simple sass -> css watcher, [which is a one-liner after installation](http://sass-lang.com/documentation/file.SASS_REFERENCE.html#using_sass).  this is a nice in-between from having them set up an entire build system for this simple page and being able to use a preprocessor that they are familiar with (and, let's be honest, is easier to look at).

noahgrant commented on README.md:
```diff
@@ -0,0 +1,57 @@
+- Finish the panel navigation so when a user clicks on a different product tab
+  the corresponding product panel appears.
+
+- Finish the Enterprise-grade panel. You may find the copy in
```
can we have this link to the part of the markup you're referring to (where you've put the comment)?

noahgrant commented on README.md:
```diff
@@ -0,0 +1,57 @@
+#### Things to get right:
+
+**HTML**
+
```
(nit: extra line, where you don't have that below)

noahgrant commented on styles/main.css:
```diff
@@ -0,0 +1,96 @@
+a {
+  text-decoration: none; }
```
nit: bracket alignment

noahgrant commented on scripts/index.js:
```diff
@@ -0,0 +1 @@
+
```
what happened to your 'add scripts here!' comment?

noahgrant commented:
question—i really like this take-home exercise, but obviously the out-of-the-box styling is pretty minimal.  do we expect them to restyle it?  if so, shouldn't we give them a finalized design?  or are we going to leave it as is on desktop and have the css they add for it be for making it responsive?

KaoruDev commented:
@garylee1 I've updated this PR to reflect the things we talked about in person. I added screenshots leading to what the final result should look like. Take a look and we can discuss more in person or via this PR. we just gotta manually delete them all haha
